### PR TITLE
Calculate Average challenge

### PR DESCRIPTION
**Pull request by Bikaze**
Implemented a solution for the challenge #40

### CODE REVIEW

File: solutions/calc_avg.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        >>> calculate_average([10, 20, 30, 40, 50])
+        30.0
+        >>> calculate_average([1.5, 2.5, 3.5])
+        2.5
+        >>> calculate_average([])
+        None
+        >>> calculate_average([100])
+        100.0
+    """
+
```
Comment (Razan-O-Elobeid) [inline solutions/calc_avg.py:37]: I suggest adding your defensive assertion here. If the list contains a non-integer or non-float element, a message should appear, such as: **"At least one element in this list is neither an integer nor a float."**

**Review by Razan-O-Elobeid — CHANGES_REQUESTED** [review]
I suggest fixing the file names to match the function name, and also review the comments for accuracy

File: solutions/tests/test_calc_avg.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+Test categories:
+    - Standard cases: multiple numbers
+    - Edge cases: empty list, single element, negative numbers, zeros
+    - Defensive tests: wrong input types, assertions
+"""
+
+import unittest
+
+from ..calc_avg import calculate_average
```
[the remark this answers is not in the document]
Comment (Bikaze) [reply solutions/tests/test_calc_avg.py:14]: Done.

File: solutions/calc_avg.py (hunk 1 above)
Comment (Bikaze) [reply solutions/calc_avg.py:37]: I added a new doctest to test the assertion.

File: solutions/tests/test_calc_avg.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Bikaze) [reply solutions/tests/test_calc_avg.py:14]: Done.

File: solutions/calc_avg.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+"""
+A module for calculating the average of a list of numbers.
+
+Module contents:
+    - calculate_average: Calculates and returns the average of a list of numbers.
+"""
+
+
+def calculate_average(numbers):
```
[the remark this answers is not in the document]
Comment (Bikaze) [reply solutions/calc_avg.py:11]: Done.

**Review by Razan-O-Elobeid — APPROVED** [review]
ok, everything is good

**Comment by Bikaze** [comment]
Thank you!
